test/methods/clipping/polygon_clipping.jl

```diff
@@ -0,0 +1,159 @@
+    [(1.0, 2.5), (3.0, 2.5), (3.0, 3.0), (1.0, 3.0), (1.0, 2.5)]])
+p43 = GI.Polygon([[(2.0, -1.0), (5.0, -1.0), (5.0, 5.0), (2.0, 5.0), (2.0, -1.0)],
+    [(3.5, 4.0), (4.5, 4.0), (4.5, 4.5), (3.5, 4.5), (3.5, 4.0)],
+    [(3.5, 3.0), (4.5, 3.0), (4.5, 3.5), (3.5, 3.5), (3.5, 3.0)],
+    [(3.5, 2.0), (4.5, 2.0), (4.5, 2.5), (3.5, 2.5), (3.5, 2.0)]])
+p44 = GI.Polygon([[(0.0, 0.0), (0.0, 10.0), (10.0, 10.0), (10.0, 0.0), (0.0, 0.0)],
+    [(5.0, 2.0), (5.0, 7.0), (7.0, 7.0), (7.0, 2.0), (5.0, 2.0)],
+    [(8.0, 2.0), (8.0, 7.0), (9.0, 7.0), (9.0, 2.0), (8.0, 2.0)],
+    [(7.25, 3.5), (7.25, 3.75), (7.75, 3.75), (7.75, 3.5), (7.25, 3.5)]])
+p45 = GI.Polygon([[(3.0, -2.0), (3.0, 8.0), (13.0, 8.0), (13.0, -2.0), (3.0, -2.0)],
+    [(5.5, 2.5), (5.5, 3.0), (8.5, 3.0), (8.5, 2.5), (5.5, 2.5)],
+    [(5.5, 4.0), (5.5, 4.5), (8.5, 4.5), (8.5, 4.0), (5.5, 4.0)]])
+
+test_pairs = [
+    (p1, p1, "p1", "p1", "Same polygon"),
+    (p1, p2, "p1", "p2", "Convex polygons that intersect (diamonds, four vertices)"),
+    (p3, p4, "p3", "p4", "Convex polygons that intersect (randomly generated, many edges)"),
+    (p5, p6, "p5", "p6", "Convex polygons that intersect (randomly generated, many edges)"),
+    (p7, p8, "p7", "p8", "Concave polygons that intersect (randomly generated, many edges)"),
+    (p5, p7, "p5", "p7", "Convex and concave polygons intersect"),
+    (p9, p10, "p9", "p10", "Figure 10 from Greiner Hormann paper"),
+    (p11, p12, "p11", "p12", "Polygons whose intersection is two distinct regions"),
+    (p13, p14, "p13", "p14", "Polygons whose intersection is two distinct regions"),
+    (p15, p16, "p15", "p16", "First polygon in second polygon"),
+    (p16, p15, "p16", "p15", "Second polygon in first polygon"),
+    (p17, p18, "p17", "p18", "First polygon with a hole (hole completly in second polygon), second without a hole"),
+    (p18, p17, "p18", "p17", "First polygon with no hole, second with a hole (hole completly in first polygon)"),
+    (p19, p20, "p19", "p20", "First polygon with a hole (hole not completly in second polygon), second without a hole"),
+    (p42, p20, "p42", "p20", "First polygon with two holes (holes not completly in second polygon), second without a hole"),
+    (p20, p19, "p20", "p19", "First polygon with no hole, second with a hole (hole not completly in first polygon)"),
+    (p20, p42, "p20", "p42", "First polygon with no holes, second with two holes (holes not completly in second polygon)"),
+    (p21, p22, "p21", "p22", "Polygons form cross, splitting each other"),
+    (p23, p24, "p23", "p24", "Polygons are both donuts with intersecting holes"),
+    (p25, p26, "p25", "p26", "Polygons both have two holes that intersect in various ways"),
+    (p27, p28, "p27", "p28", "Figure 12 from Foster extension for degeneracies"),
+    # (p29, p30, "p29", "p30", "Figure 13 from Foster extension for degeneracies"),  # will be updated to work in next PR as it has "glued edges"
```

Comment:
is this updated in this PR?

Reply:
No. That is going to be a PR that follows shortly. There are three more "extensions" in the paper that we still need to implement.